app/src/lib/git/for-each-ref.ts

```diff
@@ -101,3 +101,104 @@ export async function getBranches(
 
   return branches
 }
+
+/**
+ * Gets all branches that differ from their upstream (i.e. they're ahead,
+ * behind or both), excluding the current branch.
+ * Useful to narrow down a list of branches that could potentially be fast
+ * forwarded.
+ *
+ * @param repository Repository to get the branches from.
+ * @param allBranches All known branches in the repository.
+ * @param currentBranchName Name of the current branch.
+ */
+export async function getBranchesDifferingFromUpstream(
+  repository: Repository,
+  allBranches: ReadonlyArray<Branch>,
+  currentBranchName: string | null
```

Comment:
Not 100% confident about passing these here vs doing the actual `Branch` filtering outside, tbh. What do you think?

Reply:
I think given the name of the method I wouldn't expect it do filter out the current branch. That seems like a responsibility which should live with the caller.

I suspect tho that as we start looking at part 2 of this we could use the `%(HEAD)` format argument to `for-each-ref` to get determine if a branch is the current HEAD or not and not involve our internal list of branches at all.

Reply:
That sounds good! ❤️